app/routes/settings/users/index.js

```diff
@@ -54,12 +128,32 @@ export default class SettingsUsersIndexRoute extends Route {
     return this.store.query('user', options);
   }
 
-  setupController(controller) {
+  async setupController(controller) {
     super.setupController(...arguments);
 
     if (controller.page !== this.lastParams.committed.page) {
       controller.page = this.lastParams.committed.page;
     }
     controller.searchTextBuffer = this.lastParams.committed.filter;
+
+    const organizationIds = this.lastParams.committed.organizations;
+    controller.selectedOrganizations = await Promise.all(organizationIds.map((id) => this.store.findRecord('user-organization', id)));
+
+    controller.dateFromBuffer = parseDate(this.lastParams.committed.dateFrom);
+    controller.dateToBuffer = parseDate(this.lastParams.committed.dateTo);
+
+    const roleIds = this.lastParams.committed.roles;
+    controller.selectedRoles = await Promise.all(roleIds.map((id) => this.store.findRecord('role', id)));
```

Comment:
Page rendering will not be blocked until `setupController` completed. Maybe better to load this in the `afterModel` hook? Or define as a task on the controller that is called from here, such that we can show a loading state in the template for the `OrganizationFilter` and `UserRoleFilter` component?